src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1410,6 +1411,14 @@ internal virtual bool IsStale(QueryOperationContext queryContext, TransactionOpe
                 }
             }
 
+            if (DocumentDatabase is ShardedDocumentDatabase shardedDatabase && shardedDatabase.ShardingConfiguration.HasActiveMigrations())
```

Comment:
I think you can do that once in the `ctor`

Reply:
I assume you're referring to this part: `DocumentDatabase is ShardedDocumentDatabase shardedDatabase`?

Reply:
indeed :)

Reply:
done